### ISSUE TITLE
[UX] Convert search to a view.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1338]
Part of #1332...

Now with Views in core, we should explore the idea of taking advantage of it and convert the search to a view with exposed filters.

This would allow people to either directly edit the default *Search* view in order to tweak it to their liking or clone it and create other search views.

It would also expand the ways of extending it with contrib modules that improve the backend (search API) as well as modules that improve the UI (Views).

**Comment by jenlampton** [comment]
This may be a duplicate of https://github.com/backdrop/backdrop-issues/issues/1330, though it does have a better title :)

**Comment by klonos** [comment]
> This may be a duplicate of #1330

Yes and no... we might decide to "simply" find a way to put all search results in a single page and then provide facets or results category filtering without converting this to a view (that's basically #1330) ...or we might convert to a view and then use exposed filters etc. in order to achieve the same thing (this issue here). The result would in essence be the same, but with a non-views solution we'd have something more "hard-coded" and not as easily customizable.

**Comment by mazzech** [comment]
Hi, I would be happy to help here... but I am not sure how to make the first step. Have a lot of Search API experience in D7, but I am not a coder. But having BD search based on a View could be a good initial step, or what do you think @klonos ...?

**Comment by olafgrabienski** [comment]
I'm not sure if a view with exposed filters is better than the current core search but it would be interesting to find out. 

https://swsblog.stanford.edu/blog/better-search-drupal says:

> Faking [search] with views exposed filters is a fast, cheap, and "not a real search but sometimes good enough" (Shea McKinney) solution. If you are looking for exact keyword matching or simple filtering this may be a less resource intensive option. Views exposed filters should not be viewed as a complete search option.

@mazzech Are you maybe up to build a view with exposed filters on a content-rich Backdrop site and compare it to core search?

**Comment by jenlampton** [comment]
IIRC it's significantly slower to use views to search, than to use the search index that was specifically built for the purpose. Using the index also gives us the ability to exclude things we don't want searched (specific fields are supported in core, node types should also be, but see https://github.com/backdrop/backdrop-issues/issues/4197

**Comment by mazzech** [comment]
@olafgrabienski I never did rich content sites in Backdrop CMS so far... well, because Search API was not ready...;-) No, seriously... it's time that I contribute to Backdrop CMS, after using it on 10+ smaller sites, I will prepare some scenarios:-)
